packages/amplify-category-api/src/graphql-transformer/transformer-options-v2.ts

```diff
@@ -0,0 +1,300 @@
+import {
+  $TSAny,
+  $TSContext, $TSMeta, $TSObject,
+  AmplifyCategories,
+  AmplifySupportedService, ApiCategoryFacade,
+  CloudformationProviderFacade, getGraphQLTransformerAuthDocLink,
+  JSONUtilities,
+  pathManager,
+  stateManager,
+} from 'amplify-cli-core';
+import { AppSyncAuthConfiguration } from '@aws-amplify/graphql-transformer-interfaces';
+import {
+  collectDirectivesByTypeNames,
+} from '@aws-amplify/graphql-transformer-core';
+import {
+  getSanityCheckRules,
+  loadProject,
+} from 'graphql-transformer-core';
+import path from 'path';
+import fs from 'fs-extra';
+import { ResourceConstants } from 'graphql-transformer-common';
+import _ from 'lodash';
+import { getAdminRoles, getIdentityPoolId } from './utils';
+import { schemaHasSandboxModeEnabled, showSandboxModePrompts } from './sandbox-mode-helpers';
+import { getTransformerFactory } from './transformer-factory';
+import { AmplifyCLIFeatureFlagAdapter } from './amplify-cli-feature-flag-adapter';
+import {
+  DESTRUCTIVE_UPDATES_FLAG,
+  PARAMETERS_FILENAME,
+  PROVIDER_NAME,
+  ROOT_APPSYNC_S3_KEY,
+} from './constants';
+import {
+  TransformerFactoryArgs,
+  TransformerProjectOptions,
+} from './transformer-options-types';
+import { contextUtil } from '../category-utils/context-util';
+
+/**
+ * Use current context to generate the Transformer options for generating
+ * a GraphQL Transformer V2 object
+ * @param context The $TSContext from the Amplify CLI
+ * @param options The $TSAny options config coming from the Amplify CLI
+ */
+export const generateTransformerOptions = async (
+  context: $TSContext,
+  options: $TSAny,
+): Promise<TransformerProjectOptions<TransformerFactoryArgs>> => {
+  let resourceName: string;
+  const backEndDir = pathManager.getBackendDirPath();
+  const flags = context.parameters.options;
+  if (flags['no-gql-override']) {
+    return undefined;
+  }
+
+  let { parameters } = options;
+  const { forceCompile } = options;
+
+  // Compilation during the push step
+  const { resourcesToBeCreated, resourcesToBeUpdated, allResources } = await context.amplify.getResourceStatus(AmplifyCategories.API);
+  let resources = resourcesToBeCreated.concat(resourcesToBeUpdated);
```

Comment:
what if there are no resources to be created? Does this handles that case?

Reply:
yeah the utility function returns an empty list in that case, but it's still a list and we can use concat